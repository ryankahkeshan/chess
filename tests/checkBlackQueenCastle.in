setup   
+ K e1 + k e8
+ B g4
+ R h8 + R b4 + R c3
+ r a8
+ b f8 + b e3
+ n h7 + n a4 + n f4
+ p g7
done game human human
move g4 h5

move e8 f7

move e8 b8

move e8 c8
move g7 g6
move h5 g6
move f4 g6
move h8 f8

move e8 d8
move h7 f8
move b4 d4

move e8 c8

move e8 d8
move a4 c3
move d4 d5

move e8 c8
move c3 d5
move e1 e2
move e8 c8
