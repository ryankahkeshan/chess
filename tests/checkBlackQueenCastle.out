8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _B_
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ R
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _B_
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ R
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  R _ _B_
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ R
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  R _ _B_
3 _ R _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _k_ R
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  R _ _B_
3 _ R _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kb R
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  R _ _B_
3 _ R _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kb R
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  R _ _B_
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kb R
7 _ _ _ _n
6  _ _ _ _
5 _ _ _ _ 
4  R _ _B_
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kb R
7 _ _ _ _n
6  _ _ _ _
5 _ _ _ _ 
4 nR _ _B_
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kb R
7 _ _ _ _n
6  _ _ _ _
5 _ _ _ _ 
4 nR _ nB_
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kb R
7 _ _ _ pn
6  _ _ _ _
5 _ _ _ _ 
4 nR _ nB_
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kb R
7 _ _ _ pn
6  _ _ _ _
5 _ _ _ _ 
4 nR _ nB_
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kb R
7 _ _ _ pn
6  _ _ _ _
5 _ _ _ _B
4 nR _ n _
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Black is in check.
Invalid move
Invalid move
Invalid move
8 r_ _kb R
7 _ _ _ _n
6  _ _ _p_
5 _ _ _ _B
4 nR _ n _
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kb R
7 _ _ _ _n
6  _ _ _B_
5 _ _ _ _ 
4 nR _ n _
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Black is in check.
8 r_ _kb R
7 _ _ _ _n
6  _ _ _n_
5 _ _ _ _ 
4 nR _ _ _
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kR _
7 _ _ _ _n
6  _ _ _n_
5 _ _ _ _ 
4 nR _ _ _
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Black is in check.
Invalid move
8 r_ _kn _
7 _ _ _ _ 
6  _ _ _n_
5 _ _ _ _ 
4 nR _ _ _
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kn _
7 _ _ _ _ 
6  _ _ _n_
5 _ _ _ _ 
4 n_ R _ _
3 _ R b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Invalid move
Invalid move
8 r_ _kn _
7 _ _ _ _ 
6  _ _ _n_
5 _ _ _ _ 
4  _ R _ _
3 _ n b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kn _
7 _ _ _ _ 
6  _ _ _n_
5 _ _R_ _ 
4  _ _ _ _
3 _ n b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Invalid move
8 r_ _kn _
7 _ _ _ _ 
6  _ _ _n_
5 _ _n_ _ 
4  _ _ _ _
3 _ _ b _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8 r_ _kn _
7 _ _ _ _ 
6  _ _ _n_
5 _ _n_ _ 
4  _ _ _ _
3 _ _ b _ 
2  _ _K_ _
1 _ _ _ _ 

  abcdefgh

8  _kr n _
7 _ _ _ _ 
6  _ _ _n_
5 _ _n_ _ 
4  _ _ _ _
3 _ _ b _ 
2  _ _K_ _
1 _ _ _ _ 

  abcdefgh

Final Score:
White: 0
Black: 0
